### ISSUE TITLE
Create validate_data function for IM

### ISSUE THREAD

**Issue by ew3361zh** [RuminantFarmSystems/RuFaS#616]
Create a new function in the IM class called `validate_data` that performs validation on the data in the IM pool as instructed by the metadata.

From IM design doc:
-IM must provide data validation functionality to ensure that the structure of the data is usable and the values of the data are within the pre-defined ranges and follow specified patterns. For instance, if user wants to simulate 10^20 cows, this validation should fail as this number exceeds the maximum allowed number of cows.
-IM must terminate execution when critical inputs make validations fail. To achieve this, the function receives a flag to determine whether a validation is critical. This flag will be called `eager_termination` and will have the default value of True which means the default behavior is to terminate the process once invalid data is found.
-Minimum, maximum, minimum length, maximum length, and pattern will be used to perform validation. Since all of them are optional, the validation is contingent on their availability.

Nice to have (MVP?):
Severity value added to validation rules so some validations can be blocking and some can be non-blocking (not sure if this is planned for MVP - Sanity Check is not MVP though).
